include/awkward/builder/TupleBuilder.h

```diff
@@ -1,134 +1,110 @@
+  const std::string to_buffers(BuffersContainer &container,
+                               int64_t &form_key_id) const override;
 
-    const std::string
-      to_buffers(BuffersContainer& container, int64_t& form_key_id) const override;
+  int64_t length() const override;
 
-    int64_t
-      length() const override;
+  void clear() override;
 
-    void
-      clear() override;
-
-    /// @copydoc Builder::active()
-    ///
-    /// Calling #begintuple makes a TupleBuilder active; #endtuple makes it
-    /// inactive.
-    bool
-      active() const override;
+  /// @copydoc Builder::active()
+  ///
+  /// Calling #begintuple makes a TupleBuilder active; #endtuple makes it
+  /// inactive.
+  bool active() const override;
 
-    const BuilderPtr
-      null() override;
+  const BuilderPtr null() override;
 
-    const BuilderPtr
-      boolean(bool x) override;
+  const BuilderPtr boolean(bool x) override;
 
-    const BuilderPtr
-      integer(int64_t x) override;
+  const BuilderPtr integer(int64_t x) override;
 
-    const BuilderPtr
-      real(double x) override;
+  const BuilderPtr real(double x) override;
 
-    const BuilderPtr
-      complex(std::complex<double> x) override;
+  const BuilderPtr complex(std::complex<double> x) override;
 
-    const BuilderPtr
-      datetime(int64_t x, const std::string& unit) override;
+  const BuilderPtr datetime(int64_t x, const std::string &unit) override;
 
-    const BuilderPtr
-      timedelta(int64_t x, const std::string& unit) override;
+  const BuilderPtr timedelta(int64_t x, const std::string &unit) override;
 
-    const BuilderPtr
-      string(const char* x, int64_t length, const char* encoding) override;
+  const BuilderPtr string(const char *x, int64_t length,
+                          const char *encoding) override;
 
-    const BuilderPtr
-      beginlist() override;
+  const BuilderPtr beginlist() override;
 
-    const BuilderPtr
-      endlist() override;
+  const BuilderPtr endlist() override;
 
-    const BuilderPtr
-      begintuple(int64_t numfields) override;
+  const BuilderPtr begintuple(int64_t numfields) override;
 
-    const BuilderPtr
-      index(int64_t index) override;
+  const BuilderPtr index(int64_t index) override;
 
-    const BuilderPtr
-      endtuple() override;
+  const BuilderPtr endtuple() override;
 
-    const BuilderPtr
-      beginrecord(const char* name, bool check) override;
+  const BuilderPtr beginrecord(const char *name, bool check) override;
 
-    void
-      field(const char* key, bool check) override;
+  void field(const char *key, bool check) override;
 
-    const BuilderPtr
-      endrecord() override;
+  const BuilderPtr endrecord() override;
 
-    const BuilderOptions&
-      options() const { return options_; }
+  const BuilderOptions &options() const { return options_; }
 
-    const std::vector<BuilderPtr>& contents() const { return contents_; }
+  const std::vector<BuilderPtr> &contents() const { return contents_; }
```

Comment:
This must be that change with respect to LLVM-style that @henryiii was talking about: the `&` (and presumably also the `*`) of pointers are associated with the variable name/function name, rather than the type.

Everybody's got an opinion. _I_ think it should be associated with the type. "`int`" is one type, "`int*`" is another type, and "`int**`" is another type. If the variable names are `x`, `y`, `z`, I would write

```c++
int x;
int* y;
int** z;
```

not

```c++
int x;
int *y;
int **z;
```

The only argument I can think of for the latter has to do with how a series of variable declarations can share type descriptors:

```c++
int x, *y, **z;
```

But we shouldn't be sharing type descriptors, anyway. We should always be explicit.

So if we're getting this feature as a _deviation_ from standard LLVM style, let's go back to standard LLVM style for this one.

Reply:
This is a default option.